### PR TITLE
Updated colors and definitions

### PR DESCRIPTION
**Pull request by dempfi**
Use color definitions from ayu itself:
- Export `Color` interface for each color instead of just hex string
- Remove redundant tests. Since this package will be the main – there is no other package or definitions to tests agains
- Use TypeScript to get out of the box support for typings

### CODE REVIEW

**Comment by dempfi** [comment]
@k15a please review

File: scripts/build-svg.js [hunk 1]
```diff
@@ -35,23 +33,24 @@ let colorSVG = ''
 
 let index = 2
 for (let key in light) {
-	if (light.hasOwnProperty(key) && typeof light[key] === 'string' && light[key].length === 7) {
+	if (light.hasOwnProperty(key) && light[key] instanceof Color) {
+		// console.log(light[key])
```
Comment (k15a) [inline scripts/build-svg.js]: You can remove the console.log

File: src/colors.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import dark from './dark'
```
Comment (k15a) [inline src/colors.ts]: Maybe better if you want to support treeshaking in the future would be
```js
export {default as dark} from './dark'
export {default as light} from './light'
export {default as mirage} from './mirage'
```
But it's nothing important.

File: __tests__/colors.js [hunk 3]
```diff
@@ -1,19 +0,0 @@
...
-
-import * as colors from '../src/colors.js'
-
-const removeValues = (obj) => mapObject(obj, (key, value) => ([key, '']), { deep: true })
-
-test('colors', () => {
-	expect(colors).toMatchSnapshot()
-})
-
-test('keys equal', () => {
```
Comment (k15a) [inline __tests__/colors.js]: Maybe would be worth to keep this test so you are sure that light, dark and mirage always have the same keys. The other test is stupid and was just that I can quickly see what changed.

File: src/colors.ts (hunk 2 above)
Comment (dempfi) [reply src/colors.ts]: This syntax isn't supported by TypeScript since this syntax is still on stage 1.

**Pull request merged**